sqlx-macros-core/src/query/args.rs

```diff
@@ -89,7 +89,9 @@ pub fn quote_args<DB: DatabaseExt>(
                             _ty_check = match_borrow.match_borrow();
 
                             // this causes move-analysis to effectively ignore this block
-                            ::std::panic!();
+                            // `panic!` would be enough here but we would then get a warning on `clippy::missing_panics_doc`
+                            // SAFETY: this branch is never reached.
+                            unsafe { ::std::hint::unreachable_unchecked() }
```

Comment:
I don't like emitting `unsafe` from the macros for any reason, honestly. Does `unreachable!()` emit the same lint warning?

Reply:
This is effectively identical.
The compiler should be able to figure out that anything in `if false {...}` is unreachable ^^

Though I think clippy should also alert for this (it is still a panic just with a better name. If clippy is not, this is a false negative).

```suggestion
                            #[allow(clippy::missing_panics_doc)]
                            ::std::unreachable!();
```